site/app/libraries/database/DatabaseQueries.php

```diff
@@ -5171,6 +5187,35 @@ public function getInstructorLevelAccessCourse(string $user_id): array {
         return $this->submitty_db->rows();
     }
 
+    /**
+     * @return array<Course>
+     */
+    public function getSelfRegistrationCourses(string $user_id): array {
+        $query = <<<SQL
+SELECT c.*, t.name AS term_name FROM courses c, terms t
+WHERE c.self_registration_type > ? AND c.status = 1 and c.course NOT IN (
+    SELECT course FROM courses_users WHERE user_id = ?
+) AND c.term = t.term_id
+SQL;
+        $this->submitty_db->query($query, [ConfigurationController::NO_SELF_REGISTER, $user_id]);
+        $return = [];
+        foreach ($this->submitty_db->rows() as $row) {
+            $course = new Course($this->core, $row);
+            $course->loadDisplayName();
+            $return[] = $course;
+        }
+        return $return;
+    }
+
+    public function getSelfRegistrationType(string $course): int {
```

Comment:
Should this not also pass in the term, could a course potentially change term by term from self registration to not?

Reply:
It has been added